src/test/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusinessTest.scala

```diff
@@ -35,6 +35,29 @@ class AgoraBusinessTest extends FlatSpec with Matchers with BeforeAndAfterAll wi
 //    }
 //  }
 
+  "Agora" should "allow insertion of a method with all legal characters in its name and namespace" in {
+    val expected = patiently(agoraBusiness.insert(testAgoraEntityWithAllLegalNameChars, mockAuthenticatedOwner.get))
+    val actual = patiently(agoraBusiness.findSingle(
+      testAgoraEntityWithAllLegalNameChars,
+      Seq(testAgoraEntityWithAllLegalNameChars.entityType.get),
+      mockAuthenticatedOwner.get))
+
+    assert(actual.name == expected.name)
+    assert(actual.namespace == expected.namespace)
+  }
+
+  "Agora" should "throw an exception when inserting an entity with an illegal name" in {
+    intercept[ValidationException] {
+      patiently(agoraBusiness.insert(testAgoraEntityWithIllegalNameChars, mockAuthenticatedOwner.get))
+    }
+  }
+
+  "Agora" should "throw an exception when inserting an entity with an illegal namespace" in {
+    intercept[ValidationException] {
+      patiently(agoraBusiness.insert(testAgoraEntityWithIllegalNamespaceChars, mockAuthenticatedOwner.get))
+    }
+  }
+
```

Comment:
Idea - since Agora uses both MySQL and Mongo, there's probably a MySQL insertion required alongside Mongo to make `AgoraBusiness` `findSingle` work